storage/aptosdb/src/state_store/mod.rs

```diff
@@ -149,6 +149,12 @@ impl DbReader for StateDb {
         let mut read_opts = ReadOptions::default();
         // We want `None` if the state_key changes in iteration.
         read_opts.set_prefix_same_as_start(true);
+        // We don't need to look beyond the prune window. And it's a potential performance issue
+        // if we do -- beyond the edge of the pruning window we will see only deletions.
+        read_opts.set_iterate_upper_bound(KeyCodec::<StateValueSchema>::encode_key(&(
```

Comment:
seems the upper_bound is exclusive, should we use `get_min_readable_version - 1` here?

can we have some test to verify this works as expected?